eng/scripts/Set-CliVersionInBuild.ps1

```diff
@@ -0,0 +1,33 @@
+ <#
+    .SYNOPSIS
+    Set-CliVersionInBuild sets the CLI version defined in version.txt.
+
+    .PARAMETER BuildReason
+    The build reason supplied from the CI provider. In Azure Pipelines, this would be "Build.Reason".
+
+    .PARAMETER BuildId
+    A unique build ID supplied from the CI provider. In Azure Pipelines, this would be "Build.BuildId".
+ #>
+ param(
+    [string]$BuildReason,
+    [string]$BuildId
+ )
+
+ $prereleaseCategory = ""
+
+# TODO: Update prerelease tagging logic for GA
```

Comment:
Based on the proposed format, I believe this script does handle GA correctly. We might just tweak tag values based on requirements if needed based on #344. I don't think the TODO is necessary given the issue being tracked. @danieljurek 

```pwsh
PS D:\repos\azure-dev> Get-Content .\cli\version.txt
0.1.0-beta.6
PS D:\repos\azure-dev> eng\scripts\Set-CliVersionInBuild.ps1 -BuildReason "BatchedCI" -BuildId 123 && git co -- .\cli\version.txt
Set version.txt contents to: 0.1.0-beta.6-daily.123
PS D:\repos\azure-dev> eng\scripts\Set-CliVersionInBuild.ps1 -BuildReason "PullRequest" -BuildId 123 && git co -- .\cli\version.txt
Set version.txt contents to: 0.1.0-beta.6-pr.123
PS D:\repos\azure-dev> eng\scripts\Set-CliVersionInBuild.ps1 -BuildReason "Manual" -BuildId 123 && git co -- .\cli\version.txt     
Skipping prerelease tagging for release build.

PS D:\repos\azure-dev> Get-Content .\cli\version.txt
1.0.0
PS D:\repos\azure-dev> eng\scripts\Set-CliVersionInBuild.ps1 -BuildReason "BatchedCI" -BuildId 123 && git co -- .\cli\version.txt  
Set version.txt contents to: 1.0.0-daily.123
PS D:\repos\azure-dev> eng\scripts\Set-CliVersionInBuild.ps1 -BuildReason "PullRequest" -BuildId 123 && git co -- .\cli\version.txt
Set version.txt contents to: 1.0.0-pr.123
PS D:\repos\azure-dev> eng\scripts\Set-CliVersionInBuild.ps1 -BuildReason "Manual" -BuildId 123 && git co -- .\cli\version.txt     
Skipping prerelease tagging for release build.
```

Reply:
Right... Adding the `-` in every time means we don't need to decide whether to use `-` or `.` as in `0.1.0-beta.6.daily.123` as opposed to `0.1.0-beta.6-daily.123` and we get away with not doing semver parsing :)

Reply:
Sounds good. Removed todo, but kept the item for discussion